#73 Feature: password change prompt

chrill-christella opened the pull request:
### what does this PR do?

- user is prompted to change password x amount of time

### description

- implement change password prompt

### how should it be manually tested

-  clone the repo using git clone repo's URL
- change branch git checkout ft-password-change-prompt
- create .env file and paste variables from .env.example
- run npm run dev
- make sure that you are registered 
- navigate to login and paste in your credentials and if your password has expired there will pop up an alert
- click on that alert and navigate to your email 
- check for the password expiry email
- click on it to update your password

oliviertech212 committed:
feat(signin):signin with email and password
-create login react component
-design login ui
-test component
[Finishes #69]

oliviertech212 committed:
feat(signin):signin with email and password
-add style
[WIP]

chrill-christella commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Users should be presented with an interface that prompts them to change their passwords every x amount of time](https://trello.com/c/0RP0P8wO/73-users-should-be-presented-with-an-interface-that-prompts-them-to-change-their-passwords-every-x-amount-of-time)

sostene12 reviewed (commented):
improve the PR and show means to test the feature

chrill-christella committed:
password-change-prompt:
implement  password expiry slice
implement password expiry action
implement a pop up to notify a user that the password is expired
[finishes]

wakaflorien commented on src/redux/actions/auth.loginActions.js:
```diff
@@ -0,0 +1,51 @@
+	loginSuccess,
+	loginFail,
+	passwordExpired,
+} from '../reducers/auth/loginSlice';
```
- check this import warning.

wakaflorien commented on src/redux/reducers/rootreducer.js:
```diff
@@ -1,4 +1,6 @@
 import { combineReducers } from '@reduxjs/toolkit';
+
+import loginSlice from './auth/loginSlice';
```
- check this import warning
- make sure build is passing

wakaflorien commented on src/redux/reducers/Auth/loginSlice.js:
```diff
@@ -0,0 +1,39 @@
+			state.loading = false;
+			state.user = '';
+			state.error = action.payload;
+			state.passwordExpired = true;
```
- everytime the login fails doesn't mean the password is expired. it can easily mean entered credentials are not correct.

wakaflorien commented on src/redux/reducers/Auth/loginSlice.js:
```diff
@@ -0,0 +1,39 @@
+			state.error = action.payload;
+			state.passwordExpired = true;
+		},
+		passwordExpired: (state) => {
```
- change the rest of the state.

wakaflorien reviewed (changes requested):
- coverage folder must be ignored
- branch name should follow the correct guide.

sostene12 reviewed (commented):
- I've seen the popup as your PR said, but I'm not being redirected to where i will reset the password, and i don't see the email

sostene12 commented on coverage/components/Button/GoogleButton.jsx.html:
please remove the coverage folder

sostene12 commented on coverage/coverage-final.json:
ignore the coverage folder in .gitignore
